snitch/meson.build

```diff
@@ -0,0 +1,70 @@
+version = meson.project_version()
+vers = version.split('.')
+
+git_hash = run_command(find_program('git'),'log','-1','--format=%h'
+                      ,check:true).stdout().strip()
+
+conf_data = configuration_data({
+  'PROJECT_VERSION' : version,
+  'PROJECT_VERSION_MAJOR' : vers[0],
+  'PROJECT_VERSION_MINOR' : vers[1],
+  'PROJECT_VERSION_PATCH' : vers[2],
+
+  'SNITCH_FULL_VERSION' : version + '.' + git_hash,
+
+  'SNITCH_MAX_TEST_CASES'        : get_option('max_test_cases'),
+  'SNITCH_MAX_NESTED_SECTIONS'   : get_option('max_nested_sections'),
+  'SNITCH_MAX_EXPR_LENGTH'       : get_option('max_expr_length'),
+  'SNITCH_MAX_MESSAGE_LENGTH'    : get_option('max_message_length'),
+  'SNITCH_MAX_TEST_NAME_LENGTH'  : get_option('max_test_name_length'),
+  'SNITCH_MAX_TAG_LENGTH'        : get_option('max_tag_length'),
+  'SNITCH_MAX_CAPTURES'          : get_option('max_captures'),
+  'SNITCH_MAX_CAPTURE_LENGTH'    : get_option('max_capture_length'),
+  'SNITCH_MAX_UNIQUE_TAGS'       : get_option('max_unique_tags'),
+  'SNITCH_MAX_COMMAND_LINE_ARGS' : get_option('max_command_line_args'),
```

Comment:
Maybe you could do these with a foreach loop?

```meson
foreach opt: ['max_test_cases', 'max_nested_sections', 'max_expr_length', 'max_message_length',
              'max_test_name_length', 'max_tag_length', 'max_captures', 'max_capture_length',
              'max_unique_tags', 'max_command_line_args']
    conf_data.set('SNITCH_' + opt.to_upper(), get_option(opt))
endforeach
```

Reply:
Nice idea. I'll leave it as is, even if repetitive, as it's uniform and easy to edit when all lined up.
The mix of Integer and Bool variables would take two loops to do, and the Version variables are on their own.

It could be bit nice to have a way to iterate options, say with a `get_options()` returning a set.
That wouldn't help much here though, because of those non-uniformities.